guide/content/en/guide/basics/sync_async.md

```diff
@@ -0,0 +1,175 @@
+---
+title: Sync and Async in Sanic
+---
+
+# Sync and Async in Sanic
+
+Sanic is an **asynchronous** framework - it is designed to be responsive to requests, which allows it to theoretically serve more requests, more quickly, than a synchronous framework. Other python asynchronous frameworks also share the same limitations when calling synchronous code, so the guide below will hopefully serve you if you end up in a situation using something other than Sanic in the future.
+
+There are preventable situations where this is not the case. Using synchronous functions in asynchronous code is the biggest hurdle for many Sanic adopters.
+
+## Introduction
+Understanding asynchronicity is sometimes baffling even to experienced developers. The point of this mini-guide is to help understand what happens in Sanic, why certain things might not work the way you expect, and will help walk you through some of the impact of synchronous and asynchronous code may have on your Sanic application.
+
+There are many guides to async/await available as of this writing (Dec 2023) so we will not cover everything specifically, instead exploring and reviewing what happens inside Sanic when you use non-blocking (asynchronous) and blocking (synchronous) code.
+
+## Synchronous Execution
+Let’s start with as simple an explanation as we can produce and an example. Most python code is synchronous, that is to say when something happens, it must be done in order and must complete before the python interpreter can move on. This is called **blocking** because the process is **blocked** while it waits for each statement to finish.
+
+Our first example, below, is a standard program. The ***execution flow*** can be understood by simply reading the code.
+
+Ex. 1
+```py test1.py
+#!/usr/bin/env python3.11
+import time
+from random import randrange
+
+def sync_print(text: str) -> None:
+    rand_wait: int = randrange(0,5)
+    time.sleep(rand_wait)
+    print(f'{text} ({rand_wait}s)')
```

Comment:
I suggest to take the delay as argument, rather than random which only adds to complexity here.

Reply:
I'll be using it for consistency when I get to demonstrating the async/await versions. It adds complexity, but it's also a relatively simple path to show the impact of asynchronous code. I'm putting a pin in it, but I'd like your opinion when I've got the section complete.

Reply:
Should work with fixed delays similarly for both async and sync. I hope you have read https://trio.readthedocs.io/en/stable/tutorial.html which is a particularly good introduction to async issues (even when not using Trio).

Reply:
@Tronic after sleeping on it, I'll revise to use the delay as an argument as you suggest. I don't think the complexity is inherently a problem with using random sleep times, but given how many people seem confused about async/await removing that from the process makes sense.